mountpoint-s3/src/inode.rs

```diff
@@ -846,6 +840,14 @@ impl WriteHandle {
     }
 }
 
+impl Drop for WriteHandle {
+    fn drop(&mut self) {
+        if let Err(e) = self.finish_writing() {
+            error!(error=?e, ino=self.ino, "error closing file handle");
+        }
+    }
+}
+
```

Comment:
See my other comment -- I don't like moving fallible stuff like this into `Drop` impls, it robs you of any chance to expose the error.